File: code/python/lab8.py
from math import pow
Pam,k = 400,0.75
Pc = Pam/(1+(pow(k,2)/2))
print(f"A. Percentage of the total power in the carrier is {Pc/Pam *100} % ")
print(f"B. The power in each sideband {(Pam-Pc)*0.5} KW")
print(f"C. The total power saving if the carrier and one of the sidebands are now suppressed {round(((1- ((Pam-Pc)*0.5)/Pam)*100),2)} %")

File: code/matlab/lab8.m
clc;
clear all;
close all;
Am_signal_power = 400; %unit in KW
modulation_depth = 0.75;
carrier_power_pc = (Am_signal_power)/(1+((modulation_depth)^2)/2);
fprintf('Carrier power Pc: %.2f KW\n',carrier_power_pc);

total_power = (carrier_power_pc/Am_signal_power)*100;
fprintf('a) Total power in the carrier is: %.2f%%\n',total_power)

power_in_each_sideband = (Am_signal_power-carrier_power_pc)*0.5;
fprintf('b) Power in each sideband: %.2f KW\n',power_in_each_sideband)

percentage_power = (1-(power_in_each_sideband)/Am_signal_power)*100;
fprintf('c) Percentage power saving: %.2f%%\n',percentage_power)